packages/core/saved-objects/core-saved-objects-import-export-server-internal/src/import/lib/create_saved_objects.ts

```diff
@@ -92,20 +132,24 @@ export const createSavedObjects = async <T>({
   const resolvableErrors = ['conflict', 'ambiguous_conflict', 'missing_references'];
   let expectedResults = objectsToCreate;
   if (!accumulatedErrors.some(({ error: { type } }) => resolvableErrors.includes(type))) {
-    const bulkCreateResponse = await savedObjectsClient.bulkCreate(objectsToCreate, {
-      namespace,
-      overwrite,
-      refresh,
-    });
+    const bulkCreateResponse = await savedObjectsClient.bulkCreate(
+      [...objectsToCreate, ...Array.from(legacyAliases.values())],
+      { namespace, overwrite, refresh }
+    );
     expectedResults = bulkCreateResponse.saved_objects;
   }
 
   // remap results to reflect the object IDs that were submitted for import
   // this ensures that consumers understand the results
-  const remappedResults = expectedResults.map<CreatedObject<T>>((result) => {
+  const remappedResults = expectedResults.flatMap<CreatedObject<T>>((result) => {
+    // Only keep `error` results for the legacy URL aliases to report errors.
+    if (result.type === LEGACY_URL_ALIAS_TYPE) {
+      return result.error ? [result] : [];
+    }
```

Comment:
That's my attempt of the premature performance optimization, tried to avoid looping through the list twice, but based on the previous comment we might split these calls so it won't be a problem anymore.

> then passed the legacy url alias to extractErrors more explicitly, like with an additional parameter.

++, sounds good.